ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -89,8 +96,13 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
       throw new IllegalStateException("The NetworkClient is closed.");
     }
     List<ResponseInfo> responseInfoList = new ArrayList<ResponseInfo>();
+    numPendingConnections.set(pendingRequests.size());
```

Comment:
why set this several times within this method? Just set it once after the for loop.